### DISCUSSION TITLE
What version of FINN I have to use with Vivado 2019.2 ?

### DISCUSSION THREAD

**Discussion by dianamahdi** [Xilinx/finn#962] [Q&A] [answered]
I could run the first two Jupyter Notebooks from /notebooks/end2end_example/cybersecurity (1-train-mlp-with-brevitas.ipynb and 2-import-into-finn-and-verify.ipynb) without problems (except with netron, but it's not my point). However, I could not execute correctly the last notebook (3-build-accelerator-with-finn.ipynb) and particularly the following line: build.build_dataflow_cfg(model_file, cfg_estimates).

I wonder if it is because my Vivado version's is 2019.2 and as @fpjentzsch already mentioned, the latest version of FINN could pose some problems with Vivado versions anterior to 2022.2. However, I couldn't find a way to launch the Docker of FINN with anterior versions (I tried v0.6, v0.7, v0.8, v0.8.1 and v0.9). All these versions seems to have problems with the library attrs, as mentioned previously.

Thus, my question is the following: "What version should I prefer to use with Vivado 2019.2 ? And, there is a way to solve the problem due to attrs library ?"

Thanks in advance for your reply.

PS: I do not provide errors report, because as I tried with several FINN version, I have different errors (globally all linked with the attrs library). Thus, I prefer to ask before what version of FINN I have to use with Vivado 2019.2, before sending the errors report corresponding to this version.

**Comment by auphelia** [comment, marked as answer]
Hi @dianamahdi, 
If you would like to use an older Vivado version, you might need to make certain adjustments yourself. If you are facing the attr error, you might want to cherry-pick certain commits to use with an older version of finn. 
In general, I would highly recommend to update your Vivado/Vitis installation, for most features you can use the WebPack version and we switched to Vitis HLS a while ago, 2019.2 doesn't support that.
